### ISSUE TITLE
Visual Studio startup can be negatively impacted by credential_process based profiles

### ISSUE THREAD

**Issue by awschristou** [aws/aws-toolkit-visual-studio#115]
In https://github.com/aws/aws-toolkit-visual-studio/issues/93#issuecomment-680975507 and https://github.com/aws/aws-toolkit-visual-studio/issues/93#issuecomment-680320588, @cal5barton reports that a shared credentials profile making use of `credential_process` can cause IDE-wide delays at startup. Presumably whenever the Toolkit refreshes its credentials list as well.

In this situation, the credential process is running something that is prompting the user and waiting for them to type something in.

Currently, the toolkit is unconditionally resolving all `credential_process` profiles.

Sample `config` profile entry:
```
[profile okta-prod-proxy]
credential_process = aws-okta exec okta-prod -- cmd /c C:\batch\proxy-aws-okta-result.bat
region = us-west-2
```

**Comment by awschristou** [comment]
@cal5barton I can't offer a time frame, but if we were to make some improvements in this space, would you be interested in trying out an alpha build before the change is released?

**Comment by cal5barton** [comment]
> @cal5barton I can't offer a time frame, but if we were to make some improvements in this space, would you be interested in trying out an alpha build before the change is released?

Yes I can do that.

**Comment by awschristou** [comment]
Two improvements have been made in this space...

Earlier this week we released version 1.21.0.0. This version no longer tries to resolve _all_ of the supported credentials when the Toolkit starts up, improving the startup time.

Today we released version 1.21.1.0. Normally the AWS Explorer will try to reconnect to the credentials profile previously used in your last Visual Studio session, unless the credentials require interaction (like SSO or MFA). This version will no longer automatically reconnect to credential_process based credentials as well, because these could invoke longer running processes, which impact the startup performance. Instead, the AWS Explorer will indicate that the credentials have not been connected, and provides you with a link to establish the connection.
